### PR TITLE
Add beat/process config to elastic agent diagnostics collect

### LINKED ISSUE #29898: elastic-agent diagnostics: filebeat and metricbeat configuration
The `elastic-agent diagnostics collect` command should bundle the config that it generates for underlying beat processes

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Add the configs that are output from elastic-agent inspect output to the
bundle generated by elastic-agent diagnostics collect. Files are named
as ProcID_RouteKey.

## Why is it important?

Include the config that's sent from the agent to underlying processes in the diagnostic bundles to allow us to more easily debug config errors.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] ~~I have made corresponding change to the default configuration files~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Start an agent.
Run `elastic-agent diagnostics collect` once the agent is healthy
Inspect archive's config files.

## Related issues

- Closes #29898

### CODE REVIEW

**Comment by jlind23** [comment]
@faec may help in some of your SDHs

File: x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go [hunk 1]
```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
...
+		return cfg, err
+	}
+
+	log, err := newErrorLogger()
+	if err != nil {
+		return cfg, err
+	}
+
+	// get process config - uses same approach as inspect output command
+	// does not contact server process to request configs
```
Comment (AndersonQ) [inline x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go:383]: [Nit]
```suggestion
	// Get process config - uses same approach as inspect output command.
	// Does not contact server process to request configs.
```

File: x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go [hunk 2]
```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
 	}
 	cfg.ConfigRendered = mapCFG
 
+	// Gather vars to render process config
+	isStandalone, err := isStandalone(renderedCFG)
+	if err != nil {
+		return cfg, err
```
Comment (AndersonQ) [inline x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go:369]: [Nit / personal preference]
Usually I rather return a newly created empty struct to make explicit that the returned value is useless. Besides `cfg` might have some data, which I believe, should not be used by the caller when an error happens.
```suggestion
		return AgentConfig{}, err
```

File: x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go [hunk 3]
```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
...
+	}
+
+	agentInfo, err := info.NewAgentInfo(false)
+	if err != nil {
+		return cfg, err
```
Comment (AndersonQ) [inline x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go:374]:
```suggestion
		return AgentConfig{}, err
```

File: x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go [hunk 4]
```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
...
+	}
+
+	log, err := newErrorLogger()
+	if err != nil {
+		return cfg, err
```
Comment (AndersonQ) [inline x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go:379]:
```suggestion
		return AgentConfig{}, err
```

File: x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go [hunk 5]
```diff
@@ -361,6 +363,35 @@ func gatherConfig() (AgentConfig, error) {
...
+	}
+
+	// get process config - uses same approach as inspect output command
+	// does not contact server process to request configs
+	pMap, err := getProgramsFromConfig(log, agentInfo, renderedCFG, isStandalone)
+	if err != nil {
+		return cfg, err
```
Comment (AndersonQ) [inline x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go:386]:
```suggestion
		return AgentConfig{}, err
```

**Review by AndersonQ — APPROVED** [review]

**Review by blakerouse — APPROVED** [review]
Looks good.

**Review by ph — APPROVED** [review]

**Comment by ph** [comment]
Changes LGTM, I am been trying to test locally but other stuff is block my build.

**Pull request merged**